lasagne/layers/conv.py

```diff
@@ -777,7 +795,7 @@ def convolve(self, input, **kwargs):
             filter_flip=not self.flip_filters)
         output_size = self.output_shape[2:]
         if any(s is None for s in output_size):
-            output_size = self.get_output_shape_for(input.shape)[2:]
+            output_size = self.get_output_shape_for(input.shape, True)[2:]
```

Comment:
Add here:
```python
if isinstance(self.output_size, T.TensorVariable):
    output_size = self.output_size
elif any(s is None for s in output_size):
    output_size = self.get_output_shape_for(input.shape)[2:]
```
This avoids the `symbolic` distinction in `get_output_shape_for()`.